python/packages/nisar/products/writers/BaseL2WriterSingleInput.py

```diff
@@ -42,6 +42,52 @@
 LUT_1D_RG_DATASETS_LUT_EXPANSION_ALONG_AZ_MARGIN_IN_PIXELS = 5
 
 
+def get_static_layers_data_access(static_layers_data_access_runconfig,
+                                  granule_id):
+    """
+    Read static layers data access template from the input runconfig field
+    `static_layers_data_access` and replace placeholder(s) with
+    the actual values.
+
+    Parameters
+    ----------
+    static_layers_data_access_runconfig: scalar
+        Value from runconfig field
+        `ceos_analysis_ready_data.static_layers_data_access`
+    granule_id: str
+        Product's granule ID
+
+    Returns
+    -------
+    static_layers_data_access: str
+        An URL representing the static layers data access with all
+        placeholders replaced. Returns "(NOT SPECIFIED)" if
+        `static_layers_data_access_runconfig` is None or an empty
+        string.
```

Comment:
Let's remove the unnecessary reference to the runconfig. Let's also clearly represent what the function does so that there are no surprises for the user.

```suggestion
    static_layers_data_access: str
        Copy of `template_string` with the '{granule_id}' placeholder replaced.
        If function was unable to generate this string, an error
        is logged and a default of "(NOT SPECIFIED)" is returned.
```

Reply:
I can remove the references to the runconfig, but I think it’s useful to keep the error message explicit about the runconfig `static_layers_data_access parameter`, so it’s clear where the error originates.